### PR TITLE
ENT-9744: pip requirements check with dependabot

### PR DESCRIPTION
**Pull request by 0lmi**
Changelog: None
Ticket: ENT-9744

### CODE REVIEW

**Review by Lex-2008 — DISMISSED** [review]

**Comment by 0lmi** [comment]
@cf-bottom please test this

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/9357/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9357/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9357/

**Review by craigcomstock — APPROVED** [review]

File: packaging/cfengine-nova-hub/debian/rules [hunk 1]
```diff
@@ -86,7 +86,7 @@ install: build
 
 # Make sure cfbs is available and in place
 	mkdir -p $(CURDIR)/debian/tmp/tmp/cfbs_root
-	pip3 install --root $(CURDIR)/debian/tmp/tmp/cfbs_root cfbs==3.2.7
+	pip3 install --root $(CURDIR)/debian/tmp/tmp/cfbs_root -r cfengine-nova-hub/requirements.txt
```
Comment (olehermanse) [inline packaging/cfengine-nova-hub/debian/rules:89]: Maybe?

```suggestion
	pip3 install --root $(CURDIR)/debian/tmp/tmp/cfbs_root -r $(CURDIR)/requirements.txt
```

**Comment by 0lmi** [comment]
@cf-bottom please test this

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/9358/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9358/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9358/

**Comment by olehermanse** [comment]
Looks like the fix worked for Ubuntu 20 but not RHEL 7

**Comment by 0lmi** [comment]
@cf-bottom please test this

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/9360/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9360/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9360/

**Comment by 0lmi** [comment]
@cf-bottom please test this

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/9364/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9364/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9364/

**Comment by 0lmi** [comment]
@cf-bottom please test this

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/9365/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9365/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9365/

**Comment by 0lmi** [comment]
@cf-bottom please test this

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/9366/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9366/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9366/

**Comment by 0lmi** [comment]
@cf-bottom please test this

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/9367/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9367/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9367/

**Comment by 0lmi** [comment]
With the latest changes it successfully installs the pip package for both debian and redhat, but the jobs are failing for some other reasons...

**Comment by craigcomstock** [comment]
master canaries are likely failing due to https://github.com/cfengine/nova/commit/3ceda697fa7c3303a7d21bb37343c1d3404edaf2, which should be fixed with https://github.com/cfengine/nova/pull/2010

**Comment by 0lmi** [comment]
@cf-bottom please test this

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/9375/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9375/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9375/

**Comment by 0lmi** [comment]
`swquential-tests` job failed with the following error message, which doesn't seem like related to the changes in the PR
```
19:16:41 Errors when installing 3.15.7 version:
```

**Comment by craigcomstock** [comment]
failures are all known issues, your ci build is "green enough" to proceed. :+1:

**Review by craigcomstock — APPROVED** [review]

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+version: 2
+updates:
+  - package-ecosystem: pip
+    commit-message:
+      prefix: "chore"
+    directory: /packaging/cfengine-nova-hub/
+    schedule:
+      interval: "weekly"
```
Comment (olehermanse) [inline .github/dependabot.yml:8]: @0lmi @craigcomstock @vpodzime For cfbs upgrades, it probably makes more sense for us to do it immediately after merge / nightly, not weekly. What do you think?

Comment (Lex-2008) [reply .github/dependabot.yml:8]: hmm, then it should be part of cfbs's "on publish" action to update this repo, I guess?

Comment (craigcomstock) [reply .github/dependabot.yml:8]: yeah, but this setup is stuck to a version number that a person specifies right? So I suppose when the version changes in requirements.txt then check. What do other dependabots do for frequency? Nightly does sound a bit good to catch things sooner.

Comment (olehermanse) [reply .github/dependabot.yml:8]: @craigcomstock no. The point of this is to not update requirements.txt manually. Dependabot checks at a schedule, and updates requirements.txt if there is a new version.

Comment (olehermanse) [reply .github/dependabot.yml:8]: @Lex-2008 I'm not sure how easy / difficult it is to trigger dependabot from an on-release action. Changing to "daily", here sounds easier / good enough.

**Pull request merged**